Airports/server.js

```diff
@@ -25,18 +24,22 @@ proxyRouter.get('/\*', (req, res) => {
     }
     const url = 'https://osdatahubapi.os.uk' + req.url + separator + 'key=' + key;
 
-    if(capabilitiesExpression.test(url)) {
-        // We need to intercept and re-write GetCapabilities requests, as we don't want the capabilities document to
-        // reveal the API key to the client, and we need to re-route requests back through this proxy.
-        request(url, (request, response, body) => {
+    // We need to intercept and re-write all the requests, as we don't want the WMTS capabilities document
+    // or WFS replies to reveal the API key to the client, and we need to re-route requests back through this proxy.
+    request({ url, encoding: null }, (error, response, buffer) => {
+        const contentType = response.headers['content-type'];
+        if(contentType) {
+            res.set('Content-Type', contentType);
+        }
+        if(contentType !== 'image/png') {
+            let body = buffer.toString();
             body = body.replace(keyExpression, '');
```

Comment:
This might leave behind an extra '?' or '&' which isn't captured by the regex.

Reply:
True, but an extra ? or & should be harmless, and removing a ? could break the url... if the key was the first query parameter.

Reply:
Yeah you're right, it won't break anything. I thought it might but I tested when you have '?&' or '&&', and the empty params are just ignored.

Reply:
Yeah you're right, it won't break anything. I thought it might but I tested when you have '?&' or '&&', and the empty params are just ignored.